rangeserver/src/range_manager.rs

```diff
@@ -564,10 +586,64 @@ where
                 // gets to storage directly. We should implement a memtable to allow us to release
                 // the lock sooner.
                 lock_table.release();
+                // Process transaction complete and remove the requests from the logs
+                self.prefetching_buffer
+                    .process_transaction_complete(tx.id)
+                    .await;
                 Ok(())
             }
         }
     }
+
+    /// Get from database without acquiring any locks
+    /// This is a very basic copy of the 'get' function
+    pub async fn prefetch_get(&self, key: Bytes) -> Result<Option<Bytes>, Error> {
+        let val = self
+            .storage
+            .get(self.range_id, key.clone())
+            .await
+            .map_err(Error::from_storage_error)?;
+        Ok(val)
+    }
+
+    pub async fn process_prefetch(
+        &self,
+        buffer: &Arc<PrefetchingBuffer>,
```

Comment:
the range manager already has a reference to the prefetching buffer, so this is probably not necessary?